src/issue.rs

```diff
@@ -23,9 +23,9 @@ pub struct Issue {
     pub severity: IssueSeverity,
     pub code: String,
     pub message: String,
-    pub origin: String,
-    pub from: usize,
-    pub to: usize,
+    pub origin: Option<String>,
+    pub from: Option<usize>,
+    pub to: Option<usize>,
```

Comment:
```suggestion
    pub source: Option<(String, usize, usize)>,
```

I think this is better since we can't have 1 without the others.